x/leverage/keeper/msg_server.go

```diff
@@ -540,30 +540,20 @@ func (s msgServer) GovUpdateRegistry(
 	msg *types.MsgGovUpdateRegistry,
 ) (*types.MsgGovUpdateRegistryResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	regdTkDenoms := make(map[string]bool)
-	regdSymDenoms := make(map[string]bool)
-
+	regDenoms := make(map[string]types.Token)
 	registeredTokens := s.keeper.GetAllRegisteredTokens(ctx)
 	for _, token := range registeredTokens {
-		regdTkDenoms[token.BaseDenom] = true
-		regdSymDenoms[strings.ToUpper(token.SymbolDenom)] = true
-	}
-
-	// update the token settings
-	err := s.keeper.SaveOrUpdateTokenSettingsToRegistry(ctx, msg.UpdateTokens, regdTkDenoms, regdSymDenoms, true)
-	if err != nil {
-		return nil, err
+		regDenoms[token.BaseDenom] = token
 	}
 
-	// adds the new token settings
-	err = s.keeper.SaveOrUpdateTokenSettingsToRegistry(ctx, msg.AddTokens, regdTkDenoms, regdSymDenoms, false)
+	byEmergencyGroup := !checkers.IsGovAuthority(msg.Authority)
```

Comment:
We need to add condition to check `Authoriry` address is `EmergencyGroup`  or Not